dask/dataframe/core.py

```diff
@@ -4341,7 +4341,7 @@ def apply(self, func, convert_dtype=no_default, meta=no_default, args=(), **kwds
             kwds["convert_dtype"] = convert_dtype
 
         # let pandas trigger any warnings, such as convert_dtype warning
-        self._meta_nonempty.apply(func, args=args, **kwds)
+        self._meta_nonempty.iloc[:0].apply(func, args=args, **kwds)
```

Comment:
Probably you mean this:

```suggestion
        self._meta.apply(func, args=args, **kwds)
```

`_meta` is the norm.  `_meta_nonempty` is used when we feel some need to test actual values.

Reply:
Thx, updated. I wasn't totally sure whether I am missing a scenario where meta could contain values as well, but I guess this shouldn't be hit ever?

Reply:
Meta is empty by convention.  I'm not sure if it's actually enforced.  If it isn't then we should probably enforce it at creation time rather than everywhere else (but this is super-low priority since it doesn't seem to come up I think)